pkg/endpoint/endpoint.go

```diff
@@ -1148,10 +1151,19 @@ func (e *Endpoint) SetNamedPorts(containerPorts []slim_corev1.ContainerPort) err
 	}
 	e.mutex.Lock()
 	e.k8sPorts = k8sPorts
+	e.k8sPortsSet = true
```

Comment:
This would be better done by the caller of this function, as this would be left undone if the call sites were to be optimized to not call `SetNamedPorts()` when there are none. `e.hasK8sMetadata` should be set regardless.